lib/timers.js

```diff
@@ -317,51 +322,76 @@ exports.enroll = function(item, msecs) {
  */
 
 
-exports.setTimeout = function(callback, after) {
+exports.setTimeout = function(callback, after, arg1, arg2, arg3) {
   if (typeof callback !== 'function') {
     throw new TypeError('"callback" argument must be a function');
   }
 
-  after *= 1; // coalesce to number or NaN
-
-  if (!(after >= 1 && after <= TIMEOUT_MAX)) {
-    after = 1; // schedule on next tick, follows browser behaviour
+  var len = arguments.length;
+  var args;
+  if (len === 3) {
+    args = [arg1];
+  } else if (len === 4) {
+    args = [arg1, arg2];
+  } else if (len > 4) {
+    args = [arg1, arg2, arg3];
+    for (var i = 5; i < arguments.length; i++)
```

Comment:
It probably doesn't matter much, but you could use `len` here instead of `arguments.length`.